contracts/strategies/allocation/BaseAllocationStrategy.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import "../BaseStrategy.sol";
+import "../../interfaces/IAllocationStrategy.sol";
+
+abstract contract BaseAllocationStrategy is BaseStrategy, IAllocationStrategy {
```

Comment:
Why do we need this contract?

Reply:
It was intended to keep things consistent.

All common logic across strategies lives in BaseStrategy
All common logic across allocation strategies lives in BaseAllocationStrategy
All common logic across distribution strategies lives in BaseDistributionStrategy

Reply:
I see the confusion, since the `BaseAllocationStrategy` has basically no "unique logic", but on the other hand it combines the `BaseStrategy` and the `IAllocationStrategy`

So instead of importing the `BaseStrategy` and `IAllocationStrategy`, devs only need to import the `BaseAllocationStrategy`.

So no matter which strategy you want to build, the approach is the same. Either import the `BaseAllocationStrategy` or the `BaseDistributionStrategy` and start building.